IQSS/dataverse-frontend
# 7 - Layout component

## What this PR does / why we need it:
This PR adds the Layout component to the SPA. This PR has has 2 major reasons to exist

1. The Layout is a complete component that allows us to check that the SPA Set Up is complete and it's funcional. Also, this component opens the discussion about folders structure and organisation of the code at an early stage of the MVP 
2. The Layout component has value on itself because it defines the structure for all the Dataverse pages, it defines the Header, the Footer and the body

## Which issue(s) this PR closes:

- Closes https://github.com/IQSS/dataverse-frontend/issues/7

## Special notes for your reviewer:
### Changes
- Layout.tsx component added
- Header.tsx component now is not an example for the Storybook but the real Dataverse Header
- Footer.tsx added
- Navbar.tsx added
- Bootstrap configured to work correctly with SASS
- Fixed an issue with the i18next that was responding a 404 if the browser language did not exist
- Added route.enum.ts as the enum to contain all the SPA routes
- Grid system added with the Container.tsx, Row.tsx and Col.tsx components
- Replaced the `\` by the two spaces `  ` in the README to improve the markdown

### Vite migration
This branch is previous to the Vite migration, I'll need to fix the conflicts. Please if you come from the Vite branch remember to delete node_modules and install them again and to delete the /dist folder 

### Accessibility test error with the links
![image](https://user-images.githubusercontent.com/23359572/224973221-e36edf35-105c-4a56-90c5-0c5c5efc3fb8.png)

The links styles from the Dataverse violates some accessibility rules, we need to decide whether we:

1. Fix the link styles adding the underline and changing the color of the surrounding text and the link
2. Deactivate these accessibility rules if we think they are not necessary

## Suggestions on how to test this:

This PR doesn't add any logic, it is just replicating the Dataverse styles form the Layout so I would suggest to test it in the following way:

1. Run `npm run storybook` and go to http://localhost:6006

2. Go to the Layout in the Storybook and press the Open canvas in new tab button
![image](https://user-images.githubusercontent.com/23359572/224974640-4c44e847-f8a9-45f1-955c-ae309973d5e5.png)

3. Go to Dataverse demo or Dataverse local server and compare if everything looks and behaves as expected regarding the layout (header, footer and main container)

4. Run the unit tests for the Layout with `npm run test`

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
The mockup is the original Dataverse Layout

![image](https://user-images.githubusercontent.com/23359572/224975654-c9ef99ea-4e7c-482e-adbd-c0b3149d5c38.png)


## Is there a release notes update needed for this change?:
Header and Footer added to the SPA

## Additional documentation:
- [React Bootstrap](https://react-bootstrap.netlify.app/getting-started/introduction/)
